src/Simulation/Native/build-native-simulator.ps1

```diff
@@ -10,25 +10,62 @@ if (-not (Test-Path $nativeBuild)) {
+# https://clang.llvm.org/docs/ClangCommandLineReference.html
+# https://man7.org/linux/man-pages/man1/gcc.1.html
+$SANITIZE_FLAGS=`
+    "-fsanitize=undefined " `
+    + "-fsanitize=shift -fsanitize=shift-base " `
+    + "-fsanitize=integer-divide-by-zero -fsanitize=float-divide-by-zero " `
+    + "-fsanitize=unreachable " `
+    + "-fsanitize=vla-bound -fsanitize=null -fsanitize=return " `
+    + "-fsanitize=signed-integer-overflow -fsanitize=bounds -fsanitize=alignment -fsanitize=object-size " `
+    + "-fsanitize=float-cast-overflow -fsanitize=nonnull-attribute -fsanitize=returns-nonnull-attribute -fsanitize=bool -fsanitize=enum " `
+    + "-fsanitize=vptr -fsanitize=pointer-overflow -fsanitize=builtin " `
+    + "-fsanitize=implicit-conversion -fsanitize=local-bounds -fsanitize=nullability " `
+    `
+    + "-fsanitize=address " `
+    + "-fsanitize=pointer-compare -fsanitize=pointer-subtract " `
+    + "-fsanitize-address-use-after-scope " `
+    + "-fno-omit-frame-pointer -fno-optimize-sibling-calls"
+
+    #+ "-fsanitize=unsigned-integer-overflow "      # TODO(rokuzmin, #884): Disable this option for _specific_ lines 
+                                                    # of code, but not for the whole binary.
+
 # There should be no space after -D CMAKE_BUILD_TYPE= but if we provide the environment variable inline, without
 # the space it doesn't seem to get substituted... With invalid -D CMAKE_BUILD_TYPE argument cmake silently produces
 # a DEBUG build.
 if (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
 {
-    Write-Host "On Windows build native simulator using the default C/C++ compiler (should be MSVC)"
-    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
+    $CMAKE_C_COMPILER = "-DCMAKE_C_COMPILER=clang.exe"
+    $CMAKE_CXX_COMPILER = "-DCMAKE_CXX_COMPILER=clang++.exe"
+
+    if ((!(Get-Command clang -ErrorAction SilentlyContinue) -and (choco find --idonly -l llvm) -contains "llvm") -or `
+        (Test-Path Env:/AGENT_OS)) {
+        # LLVM was installed by Chocolatey, so add the install location to the path.
+        $env:PATH = "$($env:SystemDrive)\Program Files\LLVM\bin;$env:Path"
+    }
+
+    cmake -G Ninja -D BUILD_SHARED_LIBS:BOOL="1" $CMAKE_C_COMPILER $CMAKE_CXX_COMPILER `
+        -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION"  ..
+        # Without `-G Ninja` we fail to switch from MSVC to Clang.
+        # Sanitizers are not supported on Windows at the moment. Check again after migrating from Clang-11.
 }
 elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
 {
-    Write-Host "On Linux build native simulator using gcc (needed for OpenMP)"
-    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
+    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_C_COMPILER=clang-11 -D CMAKE_CXX_COMPILER=clang++-11 `
```

Comment:
Looks like sanitizer flags are being added even for release builds, which we don't want. The sanitizers should only be enabled for debug builds or only in the sanitizer pipeline, we don't want to ship binaries on linux and mac that have sanitizers enabled.

Reply:
Not exactly, The sanitizer flags are only added for DEBUG, see below:
CMAKE_C_FLAGS_**DEBUG**="$SANITIZE_FLAGS"
CMAKE_CXX_FLAGS_**DEBUG**="$SANITIZE_FLAGS"

Reply:
Ah, I misread, you are correct!

Reply:
Can you please uncheck the "Change requested" flag?